packages/fxa-content-server/app/scripts/models/user.js

```diff
@@ -713,6 +713,26 @@ var User = Backbone.Model.extend({
     });
   },
 
+  /**
+   * Check whether an Account's `email` is registered. Removes the account
+   * from storage if account no longer exists on the server.
+   * Additionally, retrieves third-party auth related values.
+   * @param {Object} account - account to check
+   * @returns {Promise<{
+   *  exists: boolean,
+   *  hasLinkedAccount: boolean,
+   *  hasPassword: boolean
+   * }>}
+   */
+  checkAccountStatus(account) {
```

Comment:
How big of a refactor is it to pass these options in the function above?

Reply:
@vbudhram Do you mean the function `checkAccountEmailExists` above? I created this one because 1) the name didn't seem right if checking more than just `exists`, but I could rename it across the codebase, and 2) we need a different `account.method` call if the options are needed so it's clear what we're receiving and [updating in the model](https://github.com/mozilla/fxa/pull/15456/files#diff-024557d0be847365fca264162b3fdddd6e0acd6d9b499b0adb691c2cc179b567R835-R836). 

I could rename the `checkAccountEmailExists` function, then either conditionally call `account.checkAccountStatus` if the options are passed and otherwise call `account.checkEmailExists()`, or let the conditional check happen in `account.js`. This felt more separated to match our existing patterns but I'm happy to consolidate here.

Reply:
Considering this will move to graphql when we update the frontend, I don't think it is neccessary to refactor. What you have is good 👍🏽